File: 240507/계단 수/stair-number.py
MOD = 10**9+7
n = int(input())
dp = [
    [0]*10
    for _ in range(n+1)
]
for i in range(1, 10):
    dp[1][i] = 1

for i in range(1, n+1):
    for j in range(10):
        if j < 9:
            dp[i][j] = (dp[i-1][j+1]+dp[i][j]) % MOD
        if j > 0:
            dp[i][j] = (dp[i-1][j-1]+dp[i][j]) % MOD

ans = 0
for j in range(10):
    ans = (ans + dp[n][j]) % MOD
print(ans)

File: 260901/계단 수/stair-number.java
import java.util.*;
public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        final int MOD = 1_000_000_007;
        int[][] dp = new int[n+1][10];
        for(int i = 1; i <= 9; i++)
            dp[1][i] = 1;
        for(int i = 2; i <= n; i++){
            for(int j = 0; j <= 9; j++){
                for(int k = 0; k <= 9; k++){
                    if(Math.abs(j-k) != 1) continue;
                    dp[i][k] = (dp[i][k] + dp[i-1][j]) % MOD;
                }
            }
        }
        int ans = 0;
        for(int i = 0; i <= 9; i++)
            ans = (ans + dp[n][i]) % MOD;
        System.out.println(ans);
    }
}